purge-docker-cluster.yml

```diff
@@ -1,43 +1,447 @@
 ---
 # This playbook purges a containerized Ceph cluster
-# It removes: packages, configuration files and ALL THE DATA
+# It removes: packages, containers, configuration files and ALL THE DATA
 
-- hosts:
-  - mons
-  - osds
+- name: confirm whether user really meant to purge the cluster
+
+  hosts:
+    - localhost
+
+  vars_prompt:
+    - name: ireallymeanit
+      prompt: Are you sure you want to purge the cluster?
+      default: 'no'
+      private: no
+
+  tasks:
+  - name: exit playbook, if user did not mean to purge cluster
+    fail:
+      msg: >
+        "Exiting purge-docker-cluster playbook, cluster was NOT purged.
+         To purge the cluster, either say 'yes' on the prompt or
+         or use `-e ireallymeanit=yes` on the command line when
+         invoking the playbook"
+    when: ireallymeanit != 'yes'
+
+
+- name: retrieve variables
+
+  hosts:
+    - all
```

Comment:
Using 'all' is a bit dangerous, this will get applied to the entire infrastructure. So I think we should restrain this to `mon_group_name`, `osd_group_name`, `mds_group_name` etc...